src/store/validation/index.ts

```diff
@@ -316,6 +329,40 @@ export const validationActions: ActionTree<IValidationState, IRootState> = {
   resetValidationStatuses({commit}, {key}: {key: string}): void {
     commit('resetValidationStatuses', {key})
   },
+
+  /**
+   * Compute how many resources are invalid and key them by mode and lang indexes
+   * eg:
+   * - if we have 02 languages FR, EN, 03 modes SMS, USSD, IVR, the 1st mode has a content on the 1st language
+   * - then, we expect {"modeIndex":{"0":1,"1":2,"2":2},"langIndex":{"0":2,"1":3}}
+   *
+   * @param rootGetters
+   * @param errors
+   */
+  async computeInvalidResourcesBy({rootGetters}, errors: ErrorObject[] | null): Promise<IInvalidResourcesCounterBy> {
+    const invalidCounterByMode: {[key: number]: number} = {}
+    const invalidCounterByLang: {[key: number]: number} = {}
+    const languages = rootGetters['flow/activeFlow'].languages as ILanguage[]
+    const modes = rootGetters['flow/activeFlow'].supported_modes as SupportedMode[]
+    errors?.forEach((error, index) => {
+      // replace non-numeric characters from dataPath (which could be something like `values/1/value`)
+      const currentResourceIndex = error.dataPath.replace(/\D/g, '')
+      modes.forEach((mode: SupportedMode, modeIndex: number) => {
+        languages.forEach((lang: ILanguage, langIndex: number) => {
+          const resourceIndex: string = computeResourceIndex(langIndex, modeIndex, modes.length).toString()
+          // if resourceIndex === currentResourceIndex, isValid === true, false otherwise
+          if (resourceIndex === currentResourceIndex) {
+            set(invalidCounterByMode, modeIndex, get(invalidCounterByMode, modeIndex, 0) + 1)
+            set(invalidCounterByLang, langIndex, get(invalidCounterByLang, langIndex, 0) + 1)
```

Comment:
I found [`_.update`](https://devdocs.io/lodash~4/index#update) provides a neat way to increment the counter:

```js
update(invalidCounterByMode, modeIndex, n => n ?? 0 + 1)
```